manifest-build-tools/application/update_manifest.py

```diff
@@ -0,0 +1,416 @@
+                            action="store")
+        parser.add_argument("--manifest_repo",
+                            help="The manifest repository URL.",
+                            action="store")
+        parser.add_argument("--commit",
+                            help="OPTIONAL: The commit id to target an exact version",
+                            action="store")
+        parser.add_argument("--manifest_file",
+                            help="The target manifest file to update. Searches through all if left empty.",
+                            action="store")
+        parser.add_argument("--git-credential",
+                            help="Git URL and credentials comma separated",
+                            action="append")
+        parser.add_argument('--updated_manifest',
+                            help="file containing the name of the updated manifest",
+                            action="store")
+        parser = parser.parse_args(args)
+
+        return parser
+
+
+    def assign_args(self, args):
+        """
+        Assign args to member variables and perform checks on branch and commit-id.
+        :param args: Parsed args from the user
+        :return:
+        """
+        if args.repo:
+            self.__repo = args.repo
+        else:
+            print "\nMust specify repository url for cloning (--repo <git_url>)\n"
+
+        if args.branch:
+            self.__branch = args.branch
+            if "/" in self.__branch:
+                self.__sliced_branch = self.__branch.split("/")[-1]
+            else:
+                self.__sliced_branch = self.__branch
+        else:
+            print "\nMust specify a branch name (--branch <branch_name>)\n"
+
+        if args.manifest_repo:
+            self.__manifest_repository_url = args.manifest_repo
+        else:
+            print "\n Must specify a full repository url for retrieving <manifest>.json files\n"
+
+        if args.dryrun:
+            self.__dryrun = True
+            self.repo_operator.setup_git_dryrun(self.__dryrun)
+
+        if args.commit:
+            self.__commit = args.commit
+
+        if args.manifest_file:
+            self.__manifest_file = args.manifest_file
+
+        if args.git_credential:
+            self.__git_credentials = args.git_credential
+            self.repo_operator.setup_gitbit(credentials=self.__git_credentials)
+
+        if args.updated_manifest:
+            self.__updated_manifest = args.updated_manifest
+
+    def check_args(self):
+        """
+        Check the values given for branch and commit-id
+        """
+        if self.__branch:
+            try:
+                self.repo_operator.check_branch(self.__repo, self.__sliced_branch)
+            except RuntimeError as error:
+                self.cleanup_and_exit(error, 1)
+        if self.__commit:
+            self.__check_commit()
+
+
+    def __check_commit(self):
+        """
+        Check the format of the commit-id. It must be 40 hex characters.
```

Comment:
Yeah~
I think the possibility of collision of 5 or 7 characters is bigger than 40 characters~